app/api/fetch-email-from-source.jsx

```
@@ -1,103 +1,116 @@
 
-    if (!sourceId) {
-      return new Response(JSON.stringify({ error: "SourceId is required" }), {
-        status: 400,
-        headers: { "Content-Type": "application/json" },
-      });
+export const action = async ({ params, request }) => {
+  try {
+    const { days = 7 } = await request.json();
+    const sourceId = parseInt(params.sourceId, 10);
+    
+    if (isNaN(sourceId)) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)